user.py

```diff
@@ -1,24 +1,51 @@
 import re
+
+
 class User:
-    Groups = {' ':0,'+':1,'☆':1,'%':2,'@':3,'*':3.1,'&':4,'#':5,'~':6}
+    """ Container class for basic user information collected from rooms.
+     
+    This information consists of user.id, user.rank, and user.name. user.id is
+    a format-removed id of user.name with only a-z lowercase and 0-9 present.
+
+    user.rank contain the auth level of the user, as a single character string of
+    either ' ', +, ☆, %, @, *, &, #, or ~. Note that ☆ is only relevant for 
+    battle rooms. 
+    
+    To compare groups against each other self.Groups have the information required 
```

Comment:
I know it's in the old comment you removed, but `self.Groups` shouldn't be used, ever, over `User.Groups`